docs/NewFlux_Comentary.md

```diff
@@ -0,0 +1,252 @@
+```
+
+The semantics of this syntax are to apply three different aggregate functions to the same column named `degrees_bottom`.
+The output table will then have three columns named `min_degrees_bottom`, `max_degrees_bottom`, and `mean_degrees_bottom`.
+
+
+>NOTE: Selectors do not need their own syntax as it is easy to specify selection in the existing syntax.
+
+### Column Indexing
+
+A consequence of data being represented as a single table the `tableFind` function is not needed. As such  a user can call `getColumn` directly on a table.
+We can go one step further and use the `.` syntax to mean indexing a table for its column.
+
+For example:
+
+```
+data = @foo.bar{location, state == "CA"}
+
+data.location // a list of locations in the state of CA
+```
+
+
+## The How
+
+Now that we have an overview of the new features syntax lets discuss how I expect they can be implemented.
+Again this is just to seed the conversation and is by no means a prescription on how it must be done.
+
+
+### Single Table vs Stream of Tables
+
+The current Flux internals use the interpreter to create DAG of promises to data to construct a query spec.
+That spec is then planned etc and passed down to the execution engine.
+
+So long as we can continue to build a query spec from this new Flux API we can continue to execute the queries.
+My rough plan is that a set of new Flux functions will be added to a `table` package.
+The `table` package will be very similar to the current `universe` package, it will contain all the basic functions and the new functions as well like `aggregate`.
+
+This new `table` package will need to provide its own mechanism for producing a query spec.
+The current execution engine can handle streams of tables with heterogenous schemas.
+A homogenuos schema is a subset of functionality in a heterogeneuos schema and as such can run on the exectution engine without issue.
+We will still leverage the engines concept of groups to do aggregates etc but those details will be hidden from the user.
+
+As a rough example to illustrate how this would work the following new Flux:
+
+
+```
+@data.temperatures{location,state,degrees_bottom}
+ |> aggregate({min(degrees_bottom)}, by:["location"])
+```
+
+would generate a spec similar to what this Flux would generate
+
+```
+from(bucket:"data")
+ |> filter(fn:(r) => r._measurement == "temperatures")
+ |> group(columns:["location"])
+ |> min()
+ |> group()
+```
+
+
+### Influxdb.From is always pivoted
+
+The `influxdb.from` function will be updated to accept a `measurement` parameter.
+If it exists the spec will produce the appropriate filter.
+If it doesn't exist the behavior will not change.
+Once we have versioning we can then break the API of the function and require the measurement parameter.
+
+
+### Syntax Sugar
+
+Syntax sugar will be implemented as a kind of macro.
+By macro I mean a source to source transformation before any semantic analysis is performed.
```

Comment:
Are these `macros` going to be written in pure Flux?

Reply:
I don't think so, but I don't have a clear idea of how they would be implemented.

Reply:
I think the idea of framing this syntactic sugar as macros is potentially powerful. It prefers the versatility of the underlying language while providing a stage for more terseness and innovation. I also thinks this may potentially make this language more open source/community friendly depending on how we implement this.